include/RAJA/pattern/params/reducer.hpp

```diff
@@ -83,16 +55,31 @@ namespace detail
   // Basic Reducer
   //
   //
-  template <typename Op, typename T>
+  template <template <typename, typename, typename> class Op, typename T, typename VType, typename IndexType = RAJA::Index_type>
   struct Reducer : public ForallParamBase {
-    using op = Op;
+    using op = Op<T,T,T>;
     using value_type = T;
 
     RAJA_HOST_DEVICE Reducer() {}
-    Reducer(value_type *target_in) : target(target_in), val(op::identity()) {}
+    RAJA_HOST_DEVICE Reducer(VType *target_in) : target(target_in), val(VType{}){}
+
+    VType *target = nullptr;
+    VType val = VType{};
```

Comment:
@MrBurmark @mdavis36 I'm thinking of removing `val` here, and simply using `target` throughout this class. `val` used to be `T val = op::identity()`, and I've moved the identity initializations to the `ValOp` class instead. Would it be alright to operate directly on the `target`, which would still have an SoAPtr and all the GPU accoutrements?

I was worried it would affect `get_lambda_arg_tup()` below (when the forall is extracting types and arguments), but a preliminary experiment with using `target` there seemed to work properly.

Reply:
Sure, just have to make sure target isn't de-referenced in the wrong place.